source/userguide/deep_neural_nets.rst

```diff
@@ -0,0 +1,1848 @@
+
+The deep neural models are available both for text classification aka. sequence classification (:ref:`domain classification <domain_classification>` and :ref:`intent classification <intent_classification>`) as well as for token classification (:ref:`entity recognition <entity_recognition>`) tasks.
+The models are implemented using `Pytorch <https://pytorch.org/>`_ framework and thus need extra installations before starting to use them in your chatbot application.
+
+.. note::
+
+   Please make sure to install the Pytorch requirement by running in the shell: :code:`pip install mindmeld[torch]`.
+
+MindMeld supports the use of pretrained transformer models such as :wiki_api:`BERT <BERT_(language_model)>` through the popular `Huggingface Transformers <https://huggingface.co/docs/transformers/index>`_ library.
+Several pretrained models from their `Models Hub <https://huggingface.co/models>`_ that can be used for sequence classification or token classification can be employed in your chatbot application.
+
+.. note::
+
+   Install the extra *transformers* requirement by running in the shell: :code:`pip install mindmeld[transformers]`.
+
+Before proceeding to use the deep neural models, consider the following possible advantages and disadvantages of using them in place of traditional machine learning models.
+
+Some advantages are:
+
+- The deep models generally outperform traditional machine learning models on training sets with at least several hundreds or few thousands of queries when training from scratch, and with at least few hundreds if fine-tuning from a pretrained checkpoint.
+- The deep models require little or no feature engineering work than traditional machine learning models, because they learn the input features (such as word embeddings) compared to several hundred engineered features (n-grams, system entities, and so on) of the latter.
+- On GPU-enabled devices, the deep networks can achieve training time comparable to some of the traditional models in MindMeld.
+
+Some disadvantages are:
+
+- Training time on CPU-only machines is a lot slower than for traditional machine learning models.
+- No automated hyperparameter tuning methods like :sk_api:`sklearn.model_selection.GridSearchCV <sklearn.model_selection.GridSearchCV.html>` are available for deep neural models.
+- The deep neural models generally occupy similar or more disk storage space compared to their traditional counterparts.
+
+Parameter tuning for deep neural models is more involved than for traditional machine learning models.
+A good starting point for understanding this subject is Andrej Karpathy's `course notes <https://cs231n.github.io/neural-networks-3/#baby>`_ from the Convolutional Neural Networks for Visual Recognition course at Stanford University.
```

Comment:
Say something like - While we have good defaults for all params that works well across datasets, you can understand this better using this course.